indigo/analysis/refs/base.py

```diff
@@ -78,26 +78,36 @@ class RefsFinderSubtypesENG(BaseRefsFinder):
     # country, language, locality
     locale = (None, 'eng', None)
 
-    subtypes = [s for s in Subtype.objects.all()]
-    subtype_names = [s.name for s in subtypes]
-    subtype_abbreviations = [s.abbreviation for s in subtypes]
-
-    subtypes_string = '|'.join([re.escape(s) for s in subtype_names + subtype_abbreviations])
-    pattern_re = re.compile(
-        fr'''
-            (?P<ref>
-                (?P<subtype>{subtypes_string})\s*
-                ([nN]o\.?\s*)?
-                (?P<num>\d+)
-                (\s+of\s+|/)
-                (?P<year>\d{{4}})
-            )
-        ''', re.X | re.I)
+    def setup(self, root):
+        self.get_subtypes()
+        self.get_candidate_xpath()
+        self.get_pattern_re()
+        super().setup(root)
 
-    xpath_contains = " or ".join([f"contains(translate(., '{subtype.upper()}', '{subtype.lower()}'), "
-                                  f"'{subtype.lower()}')"
-                                  for subtype in subtype_names + subtype_abbreviations])
-    candidate_xpath = f".//text()[{xpath_contains} and not(ancestor::a:ref)]"
+    def get_subtypes(self):
```

Comment:
A style comment: methods name `get_` usually don't have side effects (eg. setting attributes), they normally just return a value. In this case, since you're setting things, a better name would be `setup_`.